bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/AbstractComponent.java

```diff
@@ -77,6 +77,7 @@ public AbstractComponent(CFactory.ComponentConfiguration componentConfiguration,
         if (groupId == null || StringUtils.isBlank(groupId)) {
             groupId = this.haID.getFallbackGroupId();
         }
+        groupId = groupId.replaceAll("[^A-Za-z0-9_-]+", "_");
```

Comment:
Not sure about this. Might cause ambiguous channel IDs. I would rather use `URLEncoder.encode` as soon as core has merged my PR and allows the percent character.

Reply:
ok

Reply:
But what about "." ?

Reply:
Hm. url encode will probably leave dots untouched that's true. Would hurt to allow it for Core IDs as well, I guess. If that's not possible it should be escaped by %DOT or so. Just replacing characters is no solution, someone could have mqtt channels like "my.channel" and "my_channel".

Reply:
According to `URLEncoder.encode` doc, it does not touch [a-zA-Z0-9._*-]. All others are replaced. So let's url encode and replace "." with "%DOT"?